File: 0904-fruit-into-baskets/0904-fruit-into-baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        count, k = Counter(), 2
        j = ans = 0
        for i, fruit in enumerate(fruits):
            k -= count[fruit] == 0
            count[fruit] += 1
            while k < 0:
                count[fruits[j]] -= 1
                k += count[fruits[j]] == 0
                j += 1
            ans = max(ans, i - j + 1)
        return ans

File: 0904-fruit-into-baskets/0904-fruit-into-baskets.js
/**
 * @param {number[]} fruits
 * @return {number}
 */
var totalFruit = function(fruits) {
  let [fruitA, fruitB, countB, currMax, ans] = [-1, -1, 0, 0, 0];
  
  for (let fruit of fruits) {
    currMax = (fruit === fruitA || fruit === fruitB) ? currMax + 1 : countB + 1;   
    fruit === fruitB ? ++countB : countB = 1;    
    if (fruit !== fruitB) [fruitA, fruitB] = [fruitB, fruit];
    
    ans = Math.max(ans, currMax);
  }
  
  return ans;
};